### PR TITLE
Add a E2E test for redaction with hidden events enabled

### PR DESCRIPTION
**Pull request by luixxiul**
This PR adds E2E test for redaction with hidden events enabled. The initial intent was to create a test for https://github.com/vector-im/element-web/issues/22922, but it seems to be no longer relevant to it.

type: task

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by dbkr — APPROVED** [review]
Thanks!

**Comment by andybalaam** [comment]
This can be merged when you're ready @dbkr now we've turned off the malfunctioning Netlify check.

File: src/components/views/right_panel/PinnedMessagesCard.tsx [hunk 1]
```diff
@@ -196,7 +196,9 @@ const PinnedMessagesCard: React.FC<IProps> = ({ room, onClose, permalinkCreator
                         <div className="mx_MessageActionBar_iconButton">
                             <ReplyIcon />
                         </div>
-                        <div className="mx_MessageActionBar_iconButton mx_MessageActionBar_optionsButton">
+                        <div className="mx_MessageActionBar_iconButton mx_MessageActionBar_optionsButton"
+                            data-testid="pinned-messages-card-button-options"
```
Comment (luixxiul) [inline src/components/views/right_panel/PinnedMessagesCard.tsx:200]: This intends to ensure that `.mx_MessageActionBar_iconButton.mx_MessageActionBar_optionsButton` on pinned messages card is not selected when it is opened on the right panel along with messages on the main timeline.

**Comment by luixxiul** [comment]
Closing as this should not be a proper solution.

**Pull request closed without merging**